r/tests/testthat/test-dplyr-lubridate.R

```diff
@@ -0,0 +1,174 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+library(lubridate)
+library(dplyr)
+
+test_date <- as.POSIXct("2017-01-01 00:00:12.3456789", tz = "")
+test_df <- tibble::tibble(date = test_date)
+
+test_that("timezone aware timestamps are not supported",{
+  
+  tz_aware_date <- as.POSIXct("2017-01-01 00:00:12.3456789", tz = "BST")
+  tz_aware_df <- tibble::tibble(date = tz_aware_date)
+  
+  expect_error(
+    Table$create(tz_aware_df) %>%
+      mutate(x = wday(date)) %>%
+      collect(),
+    "Cannot extract components from timestamp with specific timezone"
```

Comment:
(1) wouldn't they work if timezone = UTC?
(2) can you link to the JIRA here in a comment?

Reply:
> (1) wouldn't they work if timezone = UTC?

We could have made it work already for UTC, but at the moment it simply checks if "a" timezone is set in the C++ code (there is an open PR to add timezone support)

Reply:
Got it, thanks, I wasn't sure where the check was happening (and great that there's a PR that would make this restriction go away anyway)

Reply:
Now have added a link to the ticket